assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/AttachmentParser.java

```diff
@@ -0,0 +1,50 @@
+package org.sakaiproject.assignment.impl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AttachmentParser {
+
+    // Pattern is thread-safe
+    private Pattern pattern = Pattern.compile("(/content)?(?<attachment>/attachment/(?<siteId>[^/]+)/Assignments/(?<id>[^/]+)/(?<filename>.*))");
+
+    public Details parse(String attachment) {
+        Matcher matcher = pattern.matcher(attachment);
+        if (matcher.matches()) {
+            return new Details(matcher.group("attachment"), matcher.group("siteId"), matcher.group("id"), matcher.group("filename"));
+        }
+        return null;
+    }
+
+    public static class Details {
```

Comment:
Does having this as public static mean that anywhere in the code I can change the static Details class?

Reply:
It's immutable. This is just used to return multiple values (added JavaDoc to explain that)